### PR TITLE
Fragment::write honors row group size 

### LINKED ISSUE #1144: Write chunked batch at the Fragment level
To facilitate the distributed writers, we should have the fine-granularity write control at the Fragment level

Right now, the chunked stream of RecordBatch is done at the global level 

https://github.com/lancedb/lancedb/blob/f35f8e451fb6b8f83ca4bad68aefb22c1341eece/node/src/arrow.ts#L36, which only works if we have one single writer.

### PR DESCRIPTION
**Pull request by eddyxu**
Closes #1144

### CODE REVIEW

**Comment by eddyxu** [comment]
The build was failing pending a `dataset::last_manifest()` is gone.

**Comment by eddyxu** [comment]
Rebased to a working `main`.

File: rust/src/dataset/fragment.rs [hunk 1]
```diff
@@ -66,35 +68,27 @@ impl FileFragment {
...
-                writer.write(&batches).await?;
-                batches = Vec::new();
-                num_rows = 0;
-            }
-        }
 
-        if num_rows > 0 {
-            writer.write(&batches).await?;
-        };
+        let mut buffered_reader = chunk_stream(stream, params.max_rows_per_file);
```
Comment (wjones127) [inline rust/src/dataset/fragment.rs:87]: Should this be `max_rows_per_group` instead?

**Review by wjones127 — COMMENTED** [review]
Most of this looks good, but are we sure that's the right parameter to use?

File: rust/src/dataset/fragment.rs (hunk 1 above)
Comment (eddyxu) [reply rust/src/dataset/fragment.rs:87]: Oh you are right, this is a typo.

Comment (eddyxu) [reply rust/src/dataset/fragment.rs]: Fixed.

**Review by wjones127 — APPROVED** [review]
Looks good!

**Pull request merged**